### DISCUSSION TITLE
Docker Compose Configuration questions

### DISCUSSION THREAD

**Discussion by sruckh** [blackcandy-org/blackcandy#272] [Q&A] [answered]
With the docker compose configuration it is unclear the purpose of nginx.  I believe it is used as a back end to blackcandy app?  If this is true, in the config it looks like you are exposing port 80 externally to port 80 in the container.  I already have a web server running on port 80, so I would need to expose a different port.  I assume there is something hard coded in blackcandy to look for nginx on port 80, so I assume exposing a different port would need some additional configuration with blackcandy.  Is this correct?

Also the docker compose file does not expose port 3000 for the blackcandy app, so I am wondering if a "port" command should be added to the blackcandy configuration?

Finally, the media_data volume in nginx-proxy section has /media_data:/media_data.  It is not clear if that should stay that way or if exported file system is supposed to be the same as the one configured in the blackcandy section.

Thank You.

**Comment by aidewoode** [comment]
It looks like you are using the latest stable version of black candy. Is this correct? This version of black candy relies on nginx to send media file. Because if you don't use nginx to send media file, it will rely on black candy backend to send file. It's less efficient than nginx. I think nginx configuration is complex for many users. So I added nginx as decency for black candy.  

According to what you said above. If you want to change the port of nginx, just change the port mapping in docker compose file. And black candy runs on port 3000, it's configure on dockerfile, so you don't need to export that port in docker-compose file.  And for the media_data volume, nginx use it to sendfile otherwise nginx wouldn't know where the media files are .

And if you want to use black candy without nginx.  I recommend using the edge version of black candy. The nginx is optional in the edge version of black candy.

**Reply by sruckh** [reply]
If I change the port mapping on nginx how does blackcandy know which port to use?

**Comment by sruckh** [comment]
I am using the image:  ghcr.io/blackcandy-org/blackcandy:edge , that was pulled less than a week ago.

**Comment by aidewoode** [comment, marked as answer]
Black candy doesn't need to know nginx port. It's reverse. It's nginx need to know the port of  blackcandy that exposed. And nginx-proxy has the ability to know the exposed port. For more informations about [nginx-proxy](https://github.com/nginx-proxy/nginx-proxy). you should check their document

**Comment by sruckh** [comment]
Ok. Got it.  Thank you for the response.
